VRDR/DeathRecord.cs

```diff
@@ -9456,56 +9456,56 @@ public string RaceRecode40Helper
         /// <para>// Getter:</para>
-        /// <para>Console.WriteLine($"First Entity Axis Code: {ExampleDeathRecord.EntityAxisCauseOfDeath[0].Item3}");</para>
+        /// <para>Console.WriteLine($"First Entity Axis Code: {ExampleDeathRecord.EntityAxisCauseOfDeath.ElementAt(0).Code}");</para>
         /// </example>
         [Property("Entity Axis Cause of Death", Property.Types.Tuple4Arr, "Entity Axis Cause of Death", "", true, IGURL.EntityAxisCauseOfDeath, false, 50)]
         [FHIRPath("Bundle.entry.resource.where($this is Observation).where(code.coding.code=80356-9)", "")]
-        public Tuple<string, string, string, string>[] EntityAxisCauseOfDeath
+        public IEnumerable<(int LineNumber, int Position, string Code, bool ECode)> EntityAxisCauseOfDeath
         {
             get
             {
-                List<Tuple<string, string, string, string>> eac = new List<Tuple<string, string, string, string>>();
+                List<(int LineNumber, int Position, string Code, bool ECode)> eac = new List<(int LineNumber, int Position, string Code, bool ECode)>();
                 if (EntityAxisCauseOfDeathObsList != null)
                 {
                     foreach (Observation ob in EntityAxisCauseOfDeathObsList)
                     {
-                        string icd10code = "";
-                        string lineNumber = "";
-                        string position = "";
-                        string ecode = "";
-                        CodeableConcept valueCC = (CodeableConcept)ob.Value;
-                        if (valueCC != null)
-                        {
-                            icd10code = valueCC.Coding[0].Code;
-                        }
+                        int? lineNumber = null;
+                        int? position = null;
+                        string icd10code = null;
+                        bool ecode = false;
                         Observation.ComponentComponent positionComp = ob.Component.Where(c => c.Code.Coding[0].Code == "position").FirstOrDefault();
                         if (positionComp != null && positionComp.Value != null)
                         {
-                            position = positionComp.Value.ToString();
+                            position = ((Integer)positionComp.Value).Value;
                         }
                         Observation.ComponentComponent lineNumComp = ob.Component.Where(c => c.Code.Coding[0].Code == "lineNumber").FirstOrDefault();
                         if (lineNumComp != null && lineNumComp.Value != null)
                         {
-                            lineNumber = lineNumComp.Value.ToString();
+                            lineNumber = ((Integer)lineNumComp.Value).Value;
+                        }
+                        CodeableConcept valueCC = (CodeableConcept)ob.Value;
+                        if (valueCC != null && valueCC.Coding != null)
```

Comment:
Should this also check that `valueCC.Length > 0` to avoid potential IndexOutOfRangeException when accessing the contents? Or is it CodeableConcept guaranteed to always have at least one Coding (if it isn't null)?

Reply:
> is it CodeableConcept guaranteed to always have at least one Coding (if it isn't null)?
There is no guarantee.   We use codable concepts that only include the text field, for example.

Reply:
I added an additional check